packages/common/container-definitions/src/runtime.ts

```diff
@@ -34,24 +34,29 @@ import type { ILoader } from "./loader.js";
  * service.  The transition from detached to attached state is a one-way transition.
  * @public
  */
-export enum AttachState {
+export const AttachState = {
```

Comment:
Not pushing to do it here, it'd maybe be more disruptive; but I think I like `AttachStates` (in general, the plural) for the const object with the possible values, and `AttachState` (the singular) for the type that covers them all, to avoid complications during imports and writing TSDoc.

Reply:
Hmm. I can see some benefits to that pattern, though a downside is that it double the number of nominal exports, means that users have to reason about the relationship between the two (e.g. knowing that they can assign `AttachStates.Attached` to a parameter of type `AttachState`). Probably not unreasonable, and easily mitigated with docs.